src/operations/command.ts

```diff
@@ -111,7 +111,7 @@ export abstract class CommandOperation<T> extends AbstractCallbackOperation<T> {
     server: Server,
     session: ClientSession | undefined,
     cmd: Document
-  ): Promise<Document> {
+  ): Promise<any> {
     // TODO: consider making this a non-enumerable property
     this.server = server;
 
```

Comment:
I can't comment on it because it's hidden in the diff but I think the inheritence hierarchy is off.  `CommandOperation` should not inherit from `AbstractCallbackOperation` - otherwise, all `CommandOperations` will still be using a callback version of `execute.  

Here's the hierarchy as I envision it.  Is there a technical reason we diverged from this approach or was it a miss in the previous PR? 

<img width="578" alt="Screenshot 2023-07-11 at 7 21 52 AM" src="https://github.com/mongodb/node-mongodb-native/assets/23407842/211b5da7-f0aa-469b-afbd-3ccb6db1153e">

Reply:
I tried the hierarchy that you pictured a while back and it was throwing errors, so I changed it to the one I currently have. However, I just tried it again and it works perfectly fine, so I will change it back. I can't remember what the error was, but I must've changed something (in execute_operation perhaps) to fix it.

Reply:
Ok I am getting those same errors again now, just checked with lint checker. If CommandOperation inherits from AbstractOperation and not AbstractCallbackOperation, all of the unchanged operations throw an error because they don't implement the abstract execute method from AbstractOperation. We need to keep it at AbstractCallbackOperation for now so it doesn't throw these errors, then change it back to AbstractOperation once all the operations are done.

Reply:
I think it's acceptable to have duplicated logic, if it's only in two place, it's temporary, and it's logic that we most likely won't touch once this PR is merged.

What's the concern with duplicating it?

Reply:
That makes sense to me, shouldn't be a problem if it's temporary.

Reply:
I do not agree, we should not have duplicate logic that can fall out of sync, we do not know exactly when we'll have gotten to the point of being able to remove it. If we are duplicating the logic we need tests to make sure they stay the same, I think that's more effort than boxing up the logic in helpers that just proxy the arguments to the existing functionality. We have many incoming changes for the major that could effect this code path, we can avoid doing things twice even temporarily.

Reply:
From slack, let's try
- to avoid dupe code by creating an instance of CommandOperation
- we have to create a small anonymous class in order to to do so